### ISSUE TITLE
Is there a way to use freeipa as a LDAP server

### ISSUE THREAD

**Issue by alinalexandru** [ldaptools/ldaptools#39]
We are using FREEIPA as LDAP server.
The **username** attribute is named **uid** in freeipa.
I don't see any setting to change that name. It seems to be hardcoded in \LdapTools\Bundle\LdapToolsBundle\Security\User\LdapUserProvider::getLdapUser:92

**Comment by ChadSikorra** [comment]
Yeah, that should be possible. I have some directions written up for it here:

https://github.com/ldaptools/ldaptools-bundle/blob/master/Resources/doc/LDAP-Authentication-Provider.md#mapping-ldap-attributes

If you have your `ldap_type` set to openldap in your config then just replace references of ad with openldap in the steps. And in step 2 replace `username: userPrincipalName` with `username: uid`. Let me know if you run into any issues.

**Comment by alinalexandru** [comment]
The first test was with success.
Can you explain what SymfonyUser parameter is doing?

**Comment by alinalexandru** [comment]
It works on website, but I don't know how to configure it for https://github.com/lexik/LexikJWTAuthenticationBundle

I can not find a way to trigger the success_handler
```
api_login:
            pattern:  ^/api/login
            provider: chain_provider
            guard:
                authenticators:
                    - ldap_tools.security.ldap_guard_authenticator
            stateless: true
            anonymous: true
            ldap_tools_form:
                check_path:               /api/login_check
                success_handler:          lexik_jwt_authentication.handler.authentication_success
                failure_handler:          lexik_jwt_authentication.handler.authentication_failure
                require_previous_session: false
```

**Comment by ChadSikorra** [comment]
The `SymfonyUser` parameter defines the name of the object type in LdapTools. In LdapTools every LDAP object (user, group, contact, etc) can have a specific object name so you can uniquely query/create those types by referring to that name. By default when you specify `ad` or `openldap` as your LdapType it has these types defined by default:

https://github.com/ldaptools/ldaptools/blob/master/docs/en/reference/Default-Schema-Attributes.md

It is purely the LdapTools library mechanism to provide a flexible schema definition for the developer. But it can be a little confusing at first glance.

**Comment by ChadSikorra** [comment]
Are you expecting to use both the Guard and the Form based login above? I think you'd want to define one or the other. But I guess I don't know what you're trying to do there. If the login uses the guard then it wouldn't use the success handler you defined. Alternatively, you could hook into the LdapToolsBundle success login event:

https://github.com/ldaptools/ldaptools-bundle/blob/master/Resources/doc/LDAP-Authentication-Provider.md#successful-login-event

You could have a class defined as a service that takes the success handler from the other bundle and triggers it there. Kind of a weird workaround, but it was the first thing that occurred to me.

**Comment by alinalexandru** [comment]
I have created 2 listeners and used the ldap_tools_bundle.guard.login.failure and ldap_tools_bundle.guard.login.success events. That's how I managed to change the response into a JSON one for the LexikJWTAuthenticationBundle bundle
